subprojects/gradle/android/src/main/kotlin/com/avito/android/AndroidSdk.kt

```diff
@@ -1,107 +1,88 @@
 package com.avito.android
 
 import com.avito.utils.ExistingDirectory
-import com.avito.utils.ExistingFile
 import com.avito.utils.ProcessRunner
-import org.funktionale.tries.Try
 import org.gradle.api.Project
 import java.io.File
 import java.util.Properties
-import kotlin.contracts.ExperimentalContracts
-import kotlin.contracts.contract
-
-/**
- * Используется для доступа к ресурсам SDK(executables, чтение параметров) из build-скриптов
- */
-@Suppress("unused")
-val Project.androidHome: String
-    get() = System.getenv("ANDROID_HOME")
-        ?: androidHomeFromLocalProperties(rootProject.file("local.properties")) { logger.error(it) }
-        ?: error("Can't resolve ANDROID_HOME")
 
 val Project.androidSdk: AndroidSdk
     get() = AndroidSdk(this, ProcessRunner.Real())
 
 class AndroidSdk(
     private val project: Project,
     private val processRunner: ProcessRunner
-) : ApkSigner, Aapt {
+) {
 
-    @Suppress("unused")
-    private val targetSdkVersion: Int
-        get() = requireNotNull(System.getProperty("targetSdkVersion").toIntOrNull()) {
-            "targetSdkVersion system property should be set"
+    private val buildToolsVersion: String
+        get() {
+            require(project.isAndroid()) { "Trying to get android.buildToolsVersion object on non-android project: ${project.path}" }
+            return project.androidBaseExtension.buildToolsVersion
         }
 
     private val compileSdkVersion: Int
-        get() = requireNotNull(System.getProperty("compileSdkVersion").toIntOrNull()) {
-            "compileSdkVersion system property should be set"
-        }
-
-
-    private val buildToolsVersion: String
         get() {
-            val value = System.getProperty("buildToolsVersion")
-            require(value.hasContent()) { "buildToolsVersion env must be set" }
-            return value
+            require(project.isAndroid()) { "Trying to get android.compileSdkVersion object on non-android project: ${project.path}" }
+            return requireNotNull(
+                value = project.androidBaseExtension.compileSdkVersion,
```

Comment:
:+1:

Reply:
Unfortunately, it's not enough. We also use `AndroidSdk` as a common abstraction, not only for a specific Android module.
See #592